feat: rewards onboarding tour

sophieqgu opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
https://consensyssoftware.atlassian.net/browse/RWDS-268
Allow user to go through onboarding tour and opt-in rewards
<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/36827?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: Add onboarding tour for user new to Rewards

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Implements the full Rewards onboarding flow (modal steps, referral, geo checks, opt-in and account linking) with new Redux state/hooks, updated controllers/actions/services, UI integrations, and comprehensive tests.
> 
> - **Rewards UI/Onboarding**:
>   - Add guided onboarding modal (`OnboardingModal`) with steps 1–4, progress indicator, referral code input/validation, legal text, and error surfaces (`RewardsErrorBanner`, `RewardsErrorToast`).
>   - Update `RewardsPointsBalance` to show sign-up badge and trigger onboarding; add clickable `RewardsBadge`.
>   - Integrate onboarding modal into Home; remove legacy `RewardsProvider` usage.
> - **State & Hooks**:
>   - Introduce Redux slice `ui/ducks/rewards` (state, actions, selectors) for onboarding, geo metadata, candidate subscription ID, season status, and error toasts.
>   - New hooks: `useOptIn`, `useLinkAccountGroup`, `useGeoRewardsMetadata`, `useCandidateSubscriptionId` (reworked), `useSeasonStatus` (reworked), `useValidateReferralCode`; remove `useRewardsEnabled`.
> - **Background/API & Controllers**:
>   - Expose rewards methods via `metamask-controller` (opt-in, referral validation, geo metadata, opt-in status/linking).
>   - `RewardsController`: change `optIn(accounts, referralCode?)`, add multi-account/linking flow, rename `getGeoRewardsMetadata` → `getRewardsGeoMetadata`, adopt shared `RewardsGeoMetadata`/opt-in DTOs.
> - **Data Service & Constants**:
>   - Default Rewards API to PRD; centralize rewards error messages; adjust tests and endpoints.
> - **Localization & Policies**:
>   - Add comprehensive rewards/onboarding strings (en/en_GB).
>   - Update LavaMoat policies for UI usage.
> - **Misc/Config**:
>   - Wire rewards actions in `ui/store/actions.ts`; update circular-deps allowlist; remove old rewards context/exports; adjust consumers (e.g., `coin-overview`).
> - **Tests**:
>   - Extensive unit/e2e tests for controllers, services, hooks, components, and reducers.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9b7323123b5b67e11f1ce67648813c115c781378. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

sophieqgu committed:
Display points balance

Add test coverage

Add test coverage

Fix lint

Fix import error

Fix lint

Fix lint

Fix lint

Revert replacing portfolio button

Replace discover button

fix bug where rewards balance was overwriting balance perc change

to verify

feat: show points estimate in swaps / bridge

Add intro step

Add onboarding step 1

Add onboarding steps

Add onboarding steps

Rebase

Rebase

Move context to redux state

Move context to redux state

Show error toast

Link accountGroup is needed

Test and fix

Fix variable names

Show modal if not seen before

Rebase

Rebase

Add test coverage

Add test coverage

Fix lint

Fix test

Fix circular dependency

Update .madgerc

Optimize images

Reduce circular dependency

Reduce circular dependency

Fix path

Update .madgerc

Fix tests

Fix tests

Revert changes to policy

HowardBraham commented on app/images/rewards/rewards-onboarding-intro-bg.png:
This PR adds several multi-megabyte images.  I think you have to 
be in not-the-most-private mode to use Rewards, right?  If so, we should host these images on a server, and not distribute them in the Extension.

sophieqgu commented on development/circular-deps.jsonc:
Now seem to reduce to around 341 lines..is this in the normal range?

sophieqgu committed:
Fix bug

sophieqgu committed:
Update policy.json

sophieqgu committed:
Update policy.json

sophieqgu committed:
Update policy.json

sophieqgu committed:
Update policy.json

sophieqgu committed:
Update errors-after-init-opt-in-ui-state.json

VGR-GIT commented on ui/components/app/rewards/onboarding/OnboardingModal.tsx:
```diff
@@ -0,0 +1,117 @@
+  const candidateSubscriptionId = useSelector(selectCandidateSubscriptionId);
+  const dispatch = useDispatch();
+
+  const theme = useTheme();
```
just like in mobile I would have candidate subscription id be either:

- a string being an actual subscription ID
- a string being 'error'
- a string being 'retry'
- a string being 'pending'

we can then act on this state here in this component/onboarding intro component & in the rewards balance component. 

- this component & onboardingintrostep: also use useCandidateSubscriptionId (don't rely on rewards balance to exist), show appropriate message if we failed to determine candidate subscription id (see mobile as well)
- rewards balance: if pending or retry show skeleton, if error and we have no season status data show failed to load (to verify with christian/andrew), if error and we have season status data show existing data,
-

VGR-GIT reviewed (changes requested):
See comments

sophieqgu committed:
Address comments

Address comments

Address comments

Update useCandidateSubscriptionId.ts

Update messages.json

sophieqgu committed:
Fix useSeasonStatus

VGR-GIT committed:
chore: pr review 1

* change useOptIn & rewardscontroller.optin so that we can now opt in the first account of the active SRP first and then link the active account as a side effect.
* Fixed a bug in useSeasonStatus where onAuthorizationError would not get handled
* Added a type for CandidateSubscriptionId
* Added onboardingModalRendered & restricted conditions for rewards onboarding modal to prevent conflicts with other modals.
* In onboardingmodal added more specificity to rule around if(candidateSubscriptionId)
* In onboarding intro step, added the case for when candidate subscription id is error.
* in onboarding intro step, added isloading and isdisabled for ctas

HowardBraham commented on development/circular-deps.jsonc:
```diff
@@ -241,88 +222,18 @@
   ],
-  ["ui/ducks/metamask/metamask.js", "ui/selectors/selectors.js"]
+  ["ui/ducks/metamask/metamask.js", "ui/selectors/selectors.js"],
+  ["ui/selectors/multichain.ts", "ui/selectors/selectors.js"]
```
Good work reducing circular dependencies, but this is the second of two _**new**_ circular dependencies that we cannot allow.

HowardBraham replied:
Oh, you're right.  It was an existing longer dependency chain, and you shortened it.

sophieqgu committed:
Update circular dependency

sophieqgu committed:
Fix useSeasonStatus dependency

MajorLift reviewed (commented):
Policy changes look ok. Will add approval for @MetaMask/policy-reviewers after more reviewers have looked at changes under `ui/`.

VGR-GIT committed:
Merge branch 'main' into rewards-onboarding-tour

HowardBraham reviewed (approved):
Approving on behalf of circular dependency reduction
